disperser/controller/node_client_manager.go

```diff
@@ -0,0 +1,60 @@
+package controller
+
+import (
+	"fmt"
+
+	"github.com/Layr-Labs/eigenda/api/clients"
+	"github.com/Layr-Labs/eigenda/core"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	lru "github.com/hashicorp/golang-lru/v2"
+)
+
+type NodeClientManager interface {
+	GetClient(operatorID core.OperatorID, socket string) (clients.NodeClientV2, error)
+}
+
+type nodeClientManager struct {
+	nodeClients *lru.Cache[core.OperatorID, clients.NodeClientV2]
+	logger      logging.Logger
+}
+
+var _ NodeClientManager = (*nodeClientManager)(nil)
+
+func NewNodeClientManager(cacheSize int, logger logging.Logger) (*nodeClientManager, error) {
+	closeClient := func(key core.OperatorID, value clients.NodeClientV2) {
+		if err := value.Close(); err != nil {
+			logger.Error("failed to close node client", "err", err)
+		}
+	}
+	nodeClients, err := lru.NewWithEvict(cacheSize, closeClient)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create LRU cache: %w", err)
+	}
+
+	return &nodeClientManager{
+		nodeClients: nodeClients,
+		logger:      logger,
+	}, nil
+}
+
+func (m *nodeClientManager) GetClient(operatorID core.OperatorID, socket string) (clients.NodeClientV2, error) {
```

Comment:
What happens if there is a socket update? Since this is only using the operatorID as the key, if there is a new socket it won't get reflected. Wouldn't it make more sense to use the socket as the key?